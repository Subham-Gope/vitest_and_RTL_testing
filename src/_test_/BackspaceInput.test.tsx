// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import BackspaceInput from "../components/BackspaceInput";

// describe("Testing the BackspaceInput.tsx file", () => {
//   test("should handle backspace key handle", async () => {
//     const user = await userEvent.setup();

//     const { container } = render(<BackspaceInput />);

//     expect(container).toMatchSnapshot();

//     const input = screen.getByRole("textbox", { name: /enter your/i });
//     expect(input).toBeInTheDocument();

//     await user.type(input, "Test");
//     expect(input).toHaveValue("Test");
//     await user.type(input, "{backspace}{backspace}");
//     expect(input).toHaveValue("Te");
//   });
// });
