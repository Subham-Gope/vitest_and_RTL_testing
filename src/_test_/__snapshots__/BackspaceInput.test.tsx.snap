// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the BackspaceInput.tsx file > should handle backspace key handle 1`] = `
<div>
  <div>
    <label
      for="backspace-input"
    >
      Enter your desired texts
    </label>
    <input
      aria-label="Enter Your Text"
      placeholder="Enter Your Text"
      type="text"
      value=""
    />
  </div>
</div>
`;
